### PR TITLE
Restructure CAPI ProwJobs

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

Fixes https://github.com/kubernetes-sigs/cluster-api/issues/10062

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api/templates/cluster-api-periodics.yaml.tpl [hunk 1]
```diff
@@ -216,3 +216,93 @@ periodics:
     testgrid-tab-name: capi-e2e-mink8s-{{ ReplaceAll $.branch "." "-" }}
     testgrid-alert-email: sig-cluster-lifecycle-cluster-api-alerts@kubernetes.io
     testgrid-num-failures-to-alert: "4"
+{{ if eq $.branch "release-1.4" "release-1.5" "release-1.6" | not }}
```
Comment (chrischdi) [inline config/jobs/kubernetes-sigs/cluster-api/templates/cluster-api-periodics.yaml.tpl:219]: ❓ we want this job for future branches right (e.g. `release-1.7)`? If it is only there to be kept for main we should match for main.

File: config/jobs/kubernetes-sigs/cluster-api/templates/cluster-api-periodics.yaml.tpl [hunk 2]
```diff
@@ -216,3 +216,93 @@ periodics:
...
+          memory: 32Gi
+        limits:
+          cpu: 7300m
+          memory: 32Gi
+  annotations:
+    testgrid-dashboards: sig-cluster-lifecycle-cluster-api{{ if eq $.branch "main" | not -}}{{ TrimPrefix $.branch "release" }}{{- end }}
+    testgrid-tab-name: capi-e2e-conformance-{{ ReplaceAll $.branch "." "-" }}
+    testgrid-alert-email: sig-cluster-lifecycle-cluster-api-alerts@kubernetes.io
+    testgrid-num-failures-to-alert: "4"
+- name: periodic-cluster-api-e2e-conformance-ci-latest-{{ ReplaceAll $.branch "." "-" }}
```
Comment (chrischdi) [inline config/jobs/kubernetes-sigs/cluster-api/templates/cluster-api-periodics.yaml.tpl:264]: ❓ we want this job for future branches right (e.g. `release-1.7)`? If it is only there to be kept for main we should add a separate if and match for main.

File: config/jobs/kubernetes-sigs/cluster-api/templates/cluster-api-periodics.yaml.tpl (hunk 1 above)
Comment (sbueringer) [reply config/jobs/kubernetes-sigs/cluster-api/templates/cluster-api-periodics.yaml.tpl:219]: Yup!

Comment (sbueringer) [reply config/jobs/kubernetes-sigs/cluster-api/templates/cluster-api-periodics.yaml.tpl:219]: Just didn't want to backport

File: config/jobs/kubernetes-sigs/cluster-api/templates/cluster-api-periodics.yaml.tpl (hunk 2 above)
Comment (sbueringer) [reply config/jobs/kubernetes-sigs/cluster-api/templates/cluster-api-periodics.yaml.tpl:264]: Yup. Same as above

**Pull request merged**